### PR TITLE
[BugFix] Make unique/foreign key constraints case insensitive

### PR DESCRIPTION
**Pull request by LiShuMing**
For MV Rewrite's constraints:
- For non-OlapTable table names, support case-insensitive; OlapTable name does not support case-insensitive.
- For columns, always support case-insensitive.

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by imay** [comment]
Should add a test-case to cover this issue

**Comment by LiShuMing** [comment]
> Should add a test-case to cover this issue

Yep. Add it .

**Review by satanson — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/scalar/MvNormalizePredicateRule.java [hunk 1]
```diff
@@ -70,13 +70,14 @@ public int compare(ScalarOperator o1, ScalarOperator o2) {
...
                         return Integer.compare(c1.getId(), c2.getId());
+                    } else {
+                        String s1 = o1.toString();
+                        String s2 = o2.toString();
+                        String n1 = s1.replaceAll("\\d+: ", "");
+                        String n2 = s2.replaceAll("\\d+: ", "");
+                        int ret = n1.compareTo(n2);
+                        return (ret == 0) ? s1.compareTo(s2) : ret;
                     }
-                    String s1 = o1.toString();
```
Comment (mofeiatwork) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/scalar/MvNormalizePredicateRule.java:74]: what's the diff?

Comment (LiShuMing) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/scalar/MvNormalizePredicateRule.java:74]: `id`  may be multi digitals.

**Review by ABingHuang — APPROVED** [review]

**Review by packy92 — APPROVED** [review]

**Pull request merged**